salt/modules/mongodb.py

```diff
@@ -156,6 +156,36 @@ def db_remove(name, user=None, password=None, host=None, port=None, authdb=None)
     return True
 
 
+def _version(mdb):
+    return mdb.command('buildInfo')['version']
+
+
+def version(user=None, password=None, host=None, port=None, database='admin', authdb=None):
+    '''
+    Get MongoDB instance version
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' mongodb.version <user> <password> <host> <port> <database>
+    '''
+    conn = _connect(user, password, host, port, authdb=authdb)
+    if not conn:
+        return 'Failed to connect to mongo database'
```

Comment:
Oh, you're right. I didn't see that the other functions were returning this way.

Now that I look more closely, I think the correct thing to do in these situations (for both this new function, as well as the others that just return a string in this module) is to raise a `CommandExecutionError`. That way it can be caught and handled correctly by other execution module functions as well as detected by the various mongodb_* state modules.

There is a new arg called `info` that you can pass into the raised exception and whatever's in that info arg is rendered using the nested outputter in the CLI return. There are some good examples of this in the [yumpkg](https://github.com/saltstack/salt/blob/5fc4c300ede98a11308cbcc958e02665de9c83bd/salt/modules/yumpkg.py#L1493-L1499) execution module, and then the the pkg state catching and handling the exception [here](https://github.com/saltstack/salt/blob/5e16de3f0064b2fed2b970c083fee7739d1fec3b/salt/states/pkg.py#L1455-L1480), for example.